classes/submission/form/PKPSubmissionSubmitStep1Form.inc.php

```diff
@@ -262,9 +262,19 @@ function execute($args, $request) {
 			// Set user to initial author
 			$authorDao = DAORegistry::getDAO('AuthorDAO');
 			$author = $authorDao->newDataObject();
-			$author->setFirstName($user->getFirstName());
-			$author->setMiddleName($user->getMiddleName());
-			$author->setLastName($user->getLastName());
+			// if no user names exist for this submission locale,
+			// copy the names in default site primary locale for this locale as well
+			$userGivenNames = $user->getGivenName(null);
+			$userFamilyNames = $user->getFamilyName(null);
+			if (empty($userGivenNames[$this->submission->getLocale()])) {
```

Comment:
Is it possible that `$userGivenNames` comes back as `null` instead of an empty array? If so, you may run into issues with expecting an array in line 272 below. Same with `$userFamilyNames`.

Reply:
Hmmm... Currently, in OJS 3, all multilingual form fields exist in the DB too, also if empty. Thus currently it shouldn't be the problem, but I could consider it however.

Reply:
Ah ok, good to know!